packages/core-data/src/actions.js

```diff
@@ -632,6 +657,34 @@ export function* saveEditedEntityRecord( kind, name, recordId, options ) {
 	yield* saveEntityRecord( kind, name, record, options );
 }
 
+/**
+ * Adds all specified entity records to a batch, and then saves them using one or more batch requests.
+ *
+ * @param {Object[]} spec List of { kind, name, key, record } of records to save.
+ * @return {Object} Finalized batch object.
+ */
+export function* __experimentalBatchSaveEntityRecords( spec ) {
+	for ( const { kind, type, key, record } of spec ) {
+		if ( key ) {
+			dispatch( 'core' ).saveEditedEntityRecord( kind, type, key, {
+				isBatch: true,
+			} );
+		} else {
+			dispatch( 'core' ).saveEntityRecord( kind, type, record, {
+				isBatch: true,
+			} );
+		}
+	}
+	yield __unstableAwaitPromise(
+		new Promise( ( resolve ) => setTimeout( resolve, 10 ) )
```

Comment:
Why do we need a random timeout here?

Reply:
My guess is that it addresses an issue similar to what I outlined in https://github.com/WordPress/gutenberg/issues/27173#issuecomment-757598581. Rungen hands execution back to `__experimentalBatchSaveEntityRecords` which calls `processBatch` before `addToBatch` has been called.

Probably we should replace this `setTimeout` with something akin to `await waitUntilNItemsAreEnqueued( spec.length )`.